### PR TITLE
Transolver volume

### PR DESCRIPTION
**Pull request by coreyjadams**
# PhysicsNeMo Pull Request

This PR brings several updates:
- The Transolver datapipe has been simplified and largely merged with DoMINO's.  This is a great performance boost.
- The Transolver example supports volumetric training.
- The Transolver++ model is, optionally, supported.

Still minor details to wrap up:
- Add plots and training results to the README for transolver.
- Update the changelog.
- Ensure the inference_on_zarr script works on volumetric data
- Update or deprecate the inference on stl script, since we can evaluate R2, etc., right from zarr.

## Checklist

- [X] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.
- [] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [X] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/cfd/external_aerodynamics/transolver/README.md [hunk 1]
```diff
@@ -190,26 +206,91 @@ Epoch 47 Validation Average Metrics:
 
 ## Dataset Inference
 
-There are two scripts provided as inference examples - it's expected that every user's
-inference workloads are different, so these aim to cover common scenarios as examples.
+<!-- There are two scripts provided as inference examples - it's expected that every user's
+inference workloads are different, so these aim to cover common scenarios as examples. -->
+
+The validation dataset in Zarr format can be loaded, processed, and the L2
+metrics summarized in `inference_on_zarr.py`.  For surface data, this script will also
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transolver/README.md:189]: Can you add that return_mesh_neighbors should be set to true to run this?

File: examples/cfd/external_aerodynamics/transolver/src/metrics.py [hunk 2]
```diff
@@ -137,26 +152,10 @@ def metrics_fn_surface(
     l2 = l2_num / l2_denom
 
     metrics = {
-        "l2_pressure": torch.mean(l2[:, 0]),
+        "l2_pressure_surf": torch.mean(l2[:, 0]),
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transolver/src/metrics.py:155]: In the readme we need to mention that this part needs to be changed when extending to a different use case. The other way is to describe variables in config.yaml like domino and read them from there.

File: examples/cfd/external_aerodynamics/transolver/src/train.py [hunk 3]
```diff
@@ -73,29 +162,30 @@ def get_autocast_context(precision: str) -> nullcontext:
...
-    else:
-        return features, embeddings
+    if dtype is not None:
+        tensors = [t.to(dtype) for t in tensors]
+
+    return tensors
 
 
-def pad_input_for_fp8(features: torch.Tensor, embeddings: torch.Tensor) -> torch.Tensor:
+def pad_input_for_fp8(
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transolver/src/train.py:188]: Do these need to be part of train.py? Can these functions be moved to utils or part of datapipe?

File: examples/cfd/external_aerodynamics/transolver/src/train.py [hunk 4]
```diff
@@ -197,16 +303,15 @@ def train_epoch(
...
     scaler: GradScaler | None = None,
 ) -> float:
     """
     Train the model for one epoch.
 
     Args:
         dataloader: Training data loader
-        sampler (torch.utils.data.Sampler): Sampler for distributed or sequential sampling.
         model (torch.nn.Module): The neural network model to train.
+        epoch_len (int): Length of the epoch.
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transolver/src/train.py:318]: Is this number of epochs?

File: examples/cfd/external_aerodynamics/transolver/README.md [hunk 5]
```diff
@@ -190,26 +206,91 @@ Epoch 47 Validation Average Metrics:
...
+|      47 | 0.0295 |        0.0564 |       0.0896 |       0.1123 |       0.1127 |                       5.1667 |                  2.7415 |                  5.2693 |                  2.779  |       12.7713 |
+[2025-11-19 07:02:38,387][training][INFO] - R2 score for lift: 0.9807
+[2025-11-19 07:02:38,387][training][INFO] - R2 score for drag: 0.9844
+[2025-11-19 07:02:38,387][training][INFO] - Summary:
+| Batch   |   Loss |   L2 Pressure |   L2 Shear X |   L2 Shear Y |   L2 Shear Z |   Predicted Drag Coefficient |   Pred Lift Coefficient |   True Drag Coefficient |   True Lift Coefficient |   Elapsed (s) |
+|---------|--------|---------------|--------------|--------------|--------------|------------------------------|-------------------------|-------------------------|-------------------------|---------------|
+| Mean    | 0.0311 |        0.0614 |       0.0921 |        0.108 |       0.1214 |                       5.2949 |                  1.9137 |                  5.2962 |                  1.9329 |       11.4647 |
```
Comment (RishikeshRanade) [inline examples/cfd/external_aerodynamics/transolver/README.md]: Is this updated? These numbers look higher than what we discussed, right?

File: examples/cfd/external_aerodynamics/transolver/src/train.py (hunk 4 above)
Comment (coreyjadams) [reply examples/cfd/external_aerodynamics/transolver/src/train.py:318]: No it's the length of the epoch, to make sure the progress in steps is correctly printed

File: examples/cfd/external_aerodynamics/transolver/README.md (hunk 5 above)
Comment (coreyjadams) [reply examples/cfd/external_aerodynamics/transolver/README.md]: Updated to the latest for this model arch.

File: examples/cfd/external_aerodynamics/transolver/README.md (hunk 1 above)
Comment (coreyjadams) [reply examples/cfd/external_aerodynamics/transolver/README.md:189]: It does it automatically now.
